tutorials/haskell/gui-application/tutorial.md

```diff
@@ -0,0 +1,825 @@
+UI designer for GTK+ called [Glade](https://glade.gnome.org/).
+
+Glade is straightforward to install and use. Open the application, you will
+see panels with various widgets: top-level objects (such as window),
+containers, and controls (such as buttons).
+
+Short plan how to re-create our calculator form with Glade:
+
+1. Select window button on the topmost palette and the window will appear on
+   the working area.
+
+2. Enter “Calculator” in the title field.
+
+3. Don't forget to fill out the “ID” attribute of every widget, this is how
+   you will access widgets on your form in the Haskell code.
+
+4. Create grid with id `grid`. When asked about number of rows and columns,
+   choose 7 × 5. Select the “Homogeneous” check box under the “Rows” title.
+
+5. Insert entry, use “Drag and resize widgets in the workspace” button on
+   the top tool bar to make it occupy the entire top row.
+
+6. Insert buttons to match our existing design.
+
+![Building the calculator form in Glade](/tutorials/haskell/gui-application/calc-3-mini.png)
+
+(Here is a [bigger image](/tutorials/haskell/gui-application/calc-3.png).)
+
+*Hint: complete glade form of our calculator is available under the Stack
+Builders tutorial repository.*
+
+To use the form in Haskell code we need `Builder`, which lives in
+`Graphics.UI.Gtk.Builder` module. The `Builder` object helps with creating
+UI from XML files on the fly.
+
+Here is how we could use it in our program:
+
+```haskell
+  …
+  builder <- builderNew                                 -- (1)
+  builderAddFromFile builder "calc.glade"               -- (2)
+  btn0 <- builderGetObject builder castToButton "btn_0" -- (3)
+  …
+```
+
+1. We need to create a new builder.
+
+2. Load our form into it from a file. There are other options such as
+   loading form text (`String`, `Text`) and so forth, consult the docs for
+   more information.
+
+3. Now the interesting part is that we can get actual button object knowing
+   its identifier. For example here I'm retrieving button that inputs zeros,
+   it has the `"btn_0"` identifier on my form. `castToButton` casts abstract
+   representatin of an object to its typed form. There are many
+   `castToSomething` functions, one per widget (for example we have
+   `castToWindow` for windows).
+
+Having the actual object like that button or main window itself, it's easy
+to proceed just like with manually constructed form to start the main loop.
+
+## Conclusion
+
+GTK+ Haskell binding certainly can be used to create professional-looking
+user interfaces. As I hopefully showed you in this tutorial, using the
+bindings is very straightforward and doesn't require any special knowledge.
+For small forms Glade probably doesn't make much sense, but if you write
+something big, it may save you some tiresome work. Better yet, one don't
+have to be a Haskell programmer to design UI with Glade, this fact makes it
+easier to divide work between people.
```

Comment:
@mrkkrp "Better yet, **you** (or 'one doesn't')" have to be a Haskell programmer to design **a** UI with Glade~~,~~ -- (or a colon) this fact makes it..."